### PR TITLE
Transcript content page

### PR DESCRIPTION
**Pull request by 0tuedon**

### CODE REVIEW

**Comment by Emmanuel-Develops** [comment]
deployment error
```
 error: TypeError: Cannot read properties of undefined (reading 'some')
      at file:///vercel/path0/.contentlayer/.cache/v0.4.3/compiled-contentlayer-config-GA5EAJV4.mjs?x=1726599239746:65:39
      at Array.forEach (<anonymous>)
      at file:///vercel/path0/.contentlayer/.cache/v0.4.3/compiled-contentlayer-config-GA5EAJV4.mjs?x=1726599239746:64:26
      at Set.forEach (<anonymous>)
      at organizeTags (file:///vercel/path0/.contentlayer/.cache/v0.4.3/compiled-contentlayer-config-GA5EAJV4.mjs?x=1726599239746:61:11)
      at onSuccess (file:///vercel/path0/.contentlayer/.cache/v0.4.3/compiled-contentlayer-config-GA5EAJV4.mjs?x=1726599239746:179:5),
  _tag: 'SuccessCallbackError',
  toString: [Function (anonymous)],
  [Symbol()]: {
    error: TypeError: Cannot read properties of undefined (reading 'some')
        at file:///vercel/path0/.contentlayer/.cache/v0.4.3/compiled-contentlayer-config-GA5EAJV4.mjs?x=1726599239746:65:39
        at Array.forEach (<anonymous>)
        at file:///vercel/path0/.contentlayer/.cache/v0.4.3/compiled-contentlayer-config-GA5EAJV4.mjs?x=1726599239746:64:26
        at Set.forEach (<anonymous>)
        at organizeTags (file:///vercel/path0/.contentlayer/.cache/v0.4.3/compiled-contentlayer-config-GA5EAJV4.mjs?x=1726599239746:61:11)
        at onSuccess (file:///vercel/path0/.contentlayer/.cache/v0.4.3/compiled-contentlayer-config-GA5EAJV4.mjs?x=1726599239746:179:5)
  },
  [Symbol()]: [ 'error' ]
}
Error: Command "npm run build" exited with 1
```

**Comment by 0tuedon** [comment]
> deployment error
> run build on local to fix

Yh fixing that.

**Comment by Emmanuel-Develops** [comment]
Screenshot 2024-09-25 at 11 20 29 AM

deployment failed on vercel. Run build on your local

**Comment by Emmanuel-Develops** [comment]
@0tuedon, update the bdp-ui package

**Comment by Emmanuel-Develops** [comment]
Clicking on a topic in the categories section links to /speakers/{{topic_here}}
e.g clicking codex32 in category page goes to https://registry-c9xl66yjb-btc-knowledge-base.vercel.app/speakers/codex32

File: next.config.mjs [hunk 1]
```diff
@@ -10,7 +10,7 @@ const nextConfig = {
         },
         {
           source: "/transcripts",
-          destination: "/gh-pages/index.html",
+          destination: "/categories",
```
Comment (Emmanuel-Develops) [inline next.config.mjs:13]: revert this change, and keep transcripts as is.
These are rewrite rules and they do not redirect, the pattern we'd want in this case would be a redirect.

Rather than a redirect, I want `/transcripts` to be an entry point to the old build. So leave it as is
However, in the landing page, footer, and header, update all links that point to `/transcripts` to point to `/categories` instead

**Review by Emmanuel-Develops — CHANGES_REQUESTED** [review]

**Comment by 0tuedon** [comment]
Updated.

File: src/app/(explore)/speakers/page.tsx [hunk 2]
```diff
@@ -8,7 +8,7 @@ const ExplorePage = () => {
 
   return (
     <div className="flex flex-col pr-2 text-black">
-        <TranscriptContentPage header="Speakers" data={allSpeakers} description="Discover insights from key figures of the Bitcoin community."  type="alphabet" linkName="speakers"/>
+        <TranscriptContentPage header="Speakers" data={allSpeakers} description="Discover insights from key figures of the Bitcoin community."  type="alphabet" linkName="tags"/>
```
Comment (Emmanuel-Develops) [inline src/app/(explore)/speakers/page.tsx:11]: this is `src/app/(explore)/speakers/page.tsx`, linkName as "speakers" is valid

File: src/app/(explore)/categories/page.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+import React from "react";
+import TranscriptContentPage from "../../../components/explore/TranscriptContentPage";
+import allCategoriesTopic from "../../../../public/tag-data.json";
+
+const ExplorePage = () => {
+
+  return (
+    <div className="flex flex-col pr-2 text-black">
+        <TranscriptContentPage header="Categories" data={allCategoriesTopic} description="Explore the main areas of focus within the Bitcoin technical ecosystem."  type="words" linkName="speakers"/>
```
Comment (Emmanuel-Develops) [inline src/app/(explore)/categories/page.tsx]: linkName should change here to tags

**Comment by Emmanuel-Develops** [comment]
Screenshot 2024-09-25 at 3 01 36 PM

**Comment by Emmanuel-Develops** [comment]
Screenshot 2024-09-26 at 9 16 17 AM

- This width is much shorter than the alphabet grouping. Make them the same.
- The current category should be bolden, in this case "Backup and recovery"

-----
Screenshot 2024-09-26 at 9 18 07 AM

- Remove types and sources from the categories.
- The font size for those links should apply only for `2xl` breakpoint, reduce the font size for lesser breakpoints

cc: @0tuedon

File: next.config.mjs (hunk 1 above)
Comment (Emmanuel-Develops) [reply next.config.mjs:13]: This wasn't resolved

Comment (Emmanuel-Develops) [reply next.config.mjs:13]: cc: @0tuedon

**Comment by Emmanuel-Develops** [comment]
@0tuedon 
Some changes haven't been attended to. 
here: https://github.com/bitcointranscripts/registry/pull/23#pullrequestreview-2328002878

File: public/gh-pages
Comment (Emmanuel-Develops) [inline public/gh-pages:1]: Do not commit submodule updates.
The workflow will handle that on main

**Comment by Emmanuel-Develops** [comment]
Screenshot 2024-09-30 at 9 56 43 AM

On mobile screens, the selection box has some margin, while scrolling other texts wrap around the selection box.
You can use the ::before psuedo-selector to create some blank space wrapping around the selection box so the scrollable content doesn't display behind the selector

File: next.config.mjs [hunk 4]
```diff
@@ -10,7 +10,7 @@ const nextConfig = {
         },
         {
           source: "/transcripts",
-          destination: "/gh-pages/index.html",
```
Comment (Emmanuel-Develops) [inline next.config.mjs:13]: revert this change 
cc: @0tuedon

**Comment by Emmanuel-Develops** [comment]
Also, #35 has been merged, so we have some merge conflicts. You'll have to rebase  😅

**Comment by 0tuedon** [comment]
> Also, #35 has been merged, so we have some merge conflicts. You'll have to rebase 😅

tears

File: src/app/globals.css [hunk 5]
```diff
@@ -2,24 +2,87 @@
...
+
+.selection-box::before {
+  content: ""; /* This creates an empty pseudo-element */
+  position: absolute;
+  top: -10px; /* Creates space above the element */
+  left: -10px; /* Creates space to the left of the element */
+  right: -10px; /* Creates space to the right of the element */
+  bottom: -10px; /* Creates space below the element */
+  background-color: white; /* Transparent to avoid affecting visual design */
+  width: 100vw;
```
Comment (Emmanuel-Develops) [inline src/app/globals.css:86]: The `width: 100vw;` in the psuedo element causes the content page to scroll horizontally.

switch to 100% width. The main concern is padding top and bottom to cut the content prematurely before getting to the selector

**Review by Emmanuel-Develops — CHANGES_REQUESTED** [review]

File: src/app/globals.css (hunk 5 above)
Comment (0tuedon) [reply src/app/globals.css:86]: Okay, but the 10px padding does that

**Comment by Emmanuel-Develops** [comment]
Screenshot 2024-10-01 at 11 42 09 PM

Notice the difference between the blue and red sectioned parts, the pseudo element doesn't cut the scrollable content on the red section
cc: @0tuedon

File: src/app/globals.css (hunk 5 above)
Comment (Emmanuel-Develops) [reply src/app/globals.css]: alright

**Review by Emmanuel-Develops — APPROVED** [review]
